postgres/lib/connect.js

```diff
@@ -571,8 +573,172 @@ function create (pool, parentCache) {
+
+			// clean audit date to use in S3 file path
+			let cleanAuditDate = client.auditdate
+				.replace(/'/g, '')
+				.replace(/:/g, '-');
+			let s3FileName = `files/${s3prefix}/${cleanAuditDate}/${table}.csv`;
+
+			client.connect().then(
+				c => {
+					client
+						.describeTable(shortTable, schema)
+						.then(result => {
+							if (ended) {
+								c.release(true);
+								return;
+							}
+							columns = result.map(f => f.column_name);
+							myClient = c;
+
+							stream = ls.toS3(leo.configuration.resources.LeoS3, s3FileName);
+							// copyFrom uses `end` but s3 `finish` so pipe finish to end
+							stream.on('finish', () => stream.emit('end'));
+
+							stream.on('error', function(err) {
+								logger.error(`COPY error: ${err.where}`, err);
+								process.exit();
+							});
+							if (pending) {
+								pending();
+							}
+						})
+						.catch(err => {
+							logger.error(err);
+						});
+				},
+				err => {
+					logger.error(err);
+				}
+			);
+
+			let count = 0;
+
+			function nonNull(v) {
+				if (v === '' || v === null || v === undefined) {
+					return '\\N';
+				} else if (typeof v === 'string' && v.search(/\r/) !== -1) {
+					return v.replace(/\r\n?/g, '\n');
+				} else {
+					return v;
+				}
+			}
+
+			
+			return ls.pipeline(
+				csv.createWriteStream({
+					...csvopts,
+					headers: false,
+					transform: (row, done) => {
+						if (!myClient) {
+							pending = () => {
+								done(
+									null,
+									columns.map(f => nonNull(row[f]))
+								);
+							};
+						} else {
+							done(
+								null,
+								columns.map(f => nonNull(row[f]))
+							);
+						}
+					},
+				}),
+				ls.write(
+					(r, done) => {
+						count++;
+						if (count % 10000 === 0) {
+							logger.info(table + ': ' + count);
+						}
+						if (!stream.write(r)) {
+							stream.once('drain', done);
+						} else {
+							done(null);
+						}
+					},
+					done => {
+						ended = true;
+						logger.debug(table + ': stream done');
+						if (stream) {
+							stream.on('end', err => {
+								logger.debug(table + ': stream ended', err || '');
+
+								// wrap done callback to release the connection
+								function innerDone(err) {
+									myClient.release(true);
+									logger.debug(table + ': stream client released', err || '');
+									done(err);
+								}
+
+								if (err) {
+									innerDone(err);
+								} else {
+									// Once the S3 file is complete run copy to load the staging table
+									let f = columns.map(f => `"${f}"`);
+									let file = `s3://${leo.configuration.s3}/${s3FileName}`;
+									let manifest = '';
+									let role = config.loaderRole;
+									myClient.query(
+										`copy ${table} (${f}) from '${file}' ${manifest} ${role ? `credentials 'aws_iam_role=${role}'` : ''
+										} NULL AS '\\\\N' format csv DELIMITER '|' ACCEPTINVCHARS TRUNCATECOLUMNS ACCEPTANYDATE TIMEFORMAT 'YYYY-MM-DD HH:MI:SS' COMPUPDATE OFF`,
```

Comment:
Our timestamp data follows a format of YYYY-MM-DDTHH:MM:SSOF which is causing them to be ingested as NULLs.  I was able to replicate this behavior and fix ingestion by replacing the value of the TIMEFORMAT parameter of `auto`.  This had the effect of lopping off the timezone data, leaving the timestamp.